### PR TITLE
Added runtime errors when reading files.

### PR DESCRIPTION
**Pull request by OGaboriault**
# Description of the problem

- The code doesn't crash when it tries to open a file that doesn't exist in the current directory. In my opinion, this is not a feature...
- i.e., tried to insert particles with the file insertion and the simulation ran for many hours with no particles in it because it didn't crash.

# Comments

- I have put Lucka as a reviewer, because it may be a "feature" for sharp.

### CODE REVIEW

**Comment by blaisb** [comment]
I don't think that's a feature anywhere. It's just a lack of sanity checking.

File: source/core/utilities.cc [hunk 1]
```diff
@@ -258,7 +258,9 @@ fill_table_from_file(TableHandler     &table,
       myfile.close();
     }
   else
-    std::cout << "Unable to open file";
+    {
+      throw std::runtime_error("Unable to open file " + file_name + ".");
```
Comment (blaisb) [inline source/core/utilities.cc:262]: Can you use the deal.II exception instead?
https://www.dealii.org/current/doxygen/deal.II/group__Exceptions.html#ga2f31207e832b937688f75c04e939cc21

**Review by blaisb — CHANGES_REQUESTED** [review]
A quick modification

File: source/core/utilities.cc [hunk 2]
```diff
@@ -657,8 +644,8 @@ get_dimension(const std::string &file_name)
       // a carriage-return without a newline -- so the error message looks
       // like this:
       //
-      //    >.  While reading the dimension from the input file, ASPECT found a
-      //    string that can not be converted to an integer: <2
+      //    >.  While reading the dimension from the input file, ASPECT found
+      //    a string that can not be converted to an integer: <2
```
Comment (blaisb) [inline source/core/utilities.cc:648]:
```suggestion
      //    >.  While reading the dimension from the input file, Lethe found
      //    a string that can not be converted to an integer: <2
```

**Review by blaisb — APPROVED** [review]
Good to merge after CI.

**Comment by Luckabarbeau** [comment]
Nothing to add here ! seems good to me.

**Pull request merged**